### PR TITLE
Feature/011 front end dummy

### LINKED ISSUE #11: Front-end dummy

### PR DESCRIPTION
**Pull request by wengcychan**
Close #11

@pixelsnow @liocle @djames9 

I have implemented routing for single page application. The registration, login and logout should be working now.

Run make and open https://localhost/

The codespace is not working because CSRF_TRUSTED_ORIGINS is set to https://localhost only

### CODE REVIEW

File: srcs/frontend/js/route.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			return;
+
+	history.pushState(null, null, path);
+	loadPage(path);
+}
+
+// Load content based on the path
+async function loadPage(path) {
+	let route = path.endsWith('/') && path!== '/' ? path.slice(0, -1) : path;
+	const page = routes[route] || window.location.pathname.slice(1);
```
Comment (linhtng) [inline srcs/frontend/js/route.js]: check if you can pass the path directly

File: srcs/backend/accounts/urls.py [hunk 2]
```diff
@@ -5,22 +5,16 @@
...
-    # Include default auth urls.
-    path('', include('django.contrib.auth.urls')),
-    # Registration page.
-    # The pattern for the registration page matches the URL http://localhost:8000/accounts/register/
-    # and sends requests to the register() function in views
     path('register/', views.register, name='register'),
+    path('login/', views.custom_login, name='login'),
+    path('logout/', views.custom_logout, name='logout'),
+    # Include default auth urls, but override specific ones as needed.
+    path('', include('django.contrib.auth.urls')),
```
Comment (linhtng) [inline srcs/backend/accounts/urls.py]: See how to properly having custom login logout without having default auth url
i.e. remove this line

**Review by pixelsnow — CHANGES_REQUESTED** [review]
reviewed together, suggested some improvements

**Review by linhtng — APPROVED** [review]
Routing for single page application is there.
Register, login, logout kinda works. However, register and login is not working properly when you input invalid info in the form. It does nothing, while the default function of register and login are informative and telling you exactly what's wrong.
But I guess it is not the main purpose of this ticket, so approving for the routing part. The work on register and login will continue in ticket
https://github.com/BeePong/42_transcendence/issues/17#issue-2393698472
I guess.

**Comment by wengcychan** [comment]
@pixelsnow Please also add the error message in the Figma and I will include it in #17 

> Routing for single page application is there. Register, login, logout kinda works. However, register and login is not working properly when you input invalid info in the form. It does nothing, while the default function of register and login are informative and telling you exactly what's wrong. But I guess it is not the main purpose of this ticket, so approving for the routing part. The work on register and login will continue in ticket [#17 (comment)](https://github.com/BeePong/42_transcendence/issues/17#issue-2393698472) I guess.

**Review by pixelsnow — APPROVED** [review]
Corrections look good, approved

**Pull request merged**